common/src/main/java/org/opensearch/ml/common/model/QuestionAnsweringModelConfig.java

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.ml.common.model;
+
+import lombok.Builder;
+import lombok.Getter;
+import lombok.Setter;
+import org.opensearch.core.ParseField;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.xcontent.NamedXContentRegistry;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.core.xcontent.XContentParser;
+import org.opensearch.ml.common.FunctionName;
+
+import java.io.IOException;
+import java.util.Locale;
+
+import static org.opensearch.core.xcontent.XContentParserUtils.ensureExpectedToken;
+
+@Setter
+@Getter
+public class QuestionAnsweringModelConfig extends MLModelConfig {
+    public static final String PARSE_FIELD_NAME = FunctionName.QUESTION_ANSWERING.name();
+    public static final NamedXContentRegistry.Entry XCONTENT_REGISTRY = new NamedXContentRegistry.Entry(
+            QuestionAnsweringModelConfig.class,
+            new ParseField(PARSE_FIELD_NAME),
+            it -> parse(it)
+    );
+    public static final String FRAMEWORK_TYPE_FIELD = "framework_type";
+    public static final String POOLING_MODE_FIELD = "pooling_mode";
+    public static final String NORMALIZE_RESULT_FIELD = "normalize_result";
+    public static final String MODEL_MAX_LENGTH_FIELD = "model_max_length";
+
+    private final FrameworkType frameworkType;
+    private final PoolingMode poolingMode;
```

Comment:
Do we really need this?  Where do we use this?

Reply:
Actually, the model is working fine without any these fields. But all the models including QA were still using TextEmbeddingModelConfig previously. Since I created this new config for QA, I had them included just as a safe sidebut we can remove them later.